src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -155,6 +155,14 @@ def __init__(self, mgr: Any) -> None:
         self.active_timers: Dict[Tuple[str, str], List[Timer]] = {}
         self.conn_lock: Lock = Lock()  # lock to protect add/lookup db connections
 
+        # restart old schedules
```

Comment:
Is comment at line 149 related to this code fix?
```
        # TODO maybe iterate over all fs instance in fsmap and load snap dbs?
        #
        # Each db connection is now guarded by a Lock; this is required to
        # avoid concurrent DB transactions when more than one paths in a
        # file-system are scheduled at the same interval eg. 1h; without the
        # lock, there are races to use the same connection, causing  nested
        # transactions to be aborted
```

Reply:
@nmshelke yup, that's right
that `TODO` was a reminder to fix this exact issue
thanks for pointing it out
now that this PR addresses the `TODO`, I have removed the `TODO` comment
I'll update the PR in a bit